### PR TITLE
Registry v2 API util update

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: api/repository.go [hunk 1]
```diff
@@ -244,13 +217,23 @@ func (ra *RepositoryAPI) GetManifests() {
...
+	if err != nil {
+		log.Errorf("error occurred while initializing repository client for %s: %v", repoName, err)
+		ra.CustomAbort(http.StatusInternalServerError, "internal error")
+	}
+
 	item := models.RepoItem{}
 
-	_, _, payload, err := ra.registry.PullManifest(repoName, tag, registry.ManifestVersion1)
+	mediaTypes := []string{schema1.MediaTypeManifest}
+	_, _, payload, err := rc.PullManifest(tag, mediaTypes)
```
Comment (reasonerjt) [inline api/repository.go]: can we have more logs so we know the error is thrown during which action, (pull manifest, list tags...)

**Comment by tinytub** [comment]
can this PR will fix /repositories issus ?
https://github.com/vmware/harbor/issues/170

**Comment by reasonerjt** [comment]
@tinytub this is the expected behavior, that the repo_name parm has to contain the namespace of the repository

**Comment by tinytub** [comment]
sorry for my last comment
i try branch https://github.com/ywk253100/harbor/tree/registry_api_util from @ywk253100 
but https://github.com/ywk253100/harbor/tree/registry_api_util_update maybe not work fine.
I got an error like https://github.com/vmware/harbor/issues/202
and it has merged to harbor's master

**Pull request merged**